mpt_trie/src/nibbles.rs

```diff
@@ -74,11 +79,58 @@ pub enum FromHexPrefixError {
     /// The hex prefix encoding flag is invalid.
     InvalidFlags(Nibble),
 
-    #[error("Tried to convert a hex prefix byte string into `Nibbles` that was longer than 64 bytes: (length: {0}, bytes: {1})")]
+    #[error("Tried to convert a hex prefix byte string into `Nibbles` that was longer than 40 bytes: (length: {0}, bytes: {1})")]
     /// The hex prefix encoding is too large.
     TooLong(String, usize),
 }
 
+/// Error type for conversion.
+#[derive(Debug, PartialEq, Eq)]
+pub enum NibblesToTypeError {
+    /// Overflow encountered.
+    Overflow,
```

Comment:
Erroring with the value that overflowed so we can dump it somewhere might be worthwhile